### PR TITLE
Add txn root, txn name and span name attributes for trace enrichment

### PR DESCRIPTION
**Pull request by lahsivjar**
Related to: https://github.com/elastic/opentelemetry-dev/issues/304

### CODE REVIEW

File: enrichments/trace/internal/elastic/span.go [hunk 1]
```diff
@@ -176,6 +182,9 @@ func (s *spanEnrichmentContext) enrichSpan(
 	span ptrace.Span,
 	cfg config.ElasticSpanConfig,
 ) {
+	if cfg.Name.Enabled {
+		span.Attributes().PutStr(AttributeSpanName, span.Name())
```
Comment (axw) [inline enrichments/trace/internal/elastic/span.go:186]: Won't we already set `span.name` in the doc? Why is this needed?

Comment (lahsivjar) [reply enrichments/trace/internal/elastic/span.go:186]: Span name is not an attribute and we need some way to add this to the metrics produced by the connector. The connector would only process attributes and add it to the resulting metric. We could add some support with OTTL in the connector to create attributes based on expressions but I thought this would be better since it is a kind of enrichment.

Comment (lahsivjar) [reply enrichments/trace/internal/elastic/span.go:186]: For a bit more clarity, the `spanmetricsv2` connector would look like this with the current PR:

```yaml
  spanmetricsv2:
    spans:
      service_transaction:
        description: APM service transaction aggregated metrics
        attributes:
        - key: transaction.root
      transaction:
        description: APM transaction aggregated metrics
        attributes:
        - key: transaction.root
        - key: transaction.name
        - key: transaction.result
        - key: event.outcome
      span_destination:
        description: APM span destination metrics
        attributes:
          - key: span.name
          - key: event.outcome
          - key: service.target.type
          - key: service.target.name
```

Comment (axw) [reply enrichments/trace/internal/elastic/span.go:186]: IMO having OTTL support would be the ideal end state. I guess this is fine for now - maybe we can come back to it.

A couple more thoughts about OTTL:

Since we only care about having `transaction.root` on transaction metrics, and nothing else, we could use the `count` connector with an OTTL-based condition on parent_span_id being empty, and (in theory) add an attribute with config in the connector, rather than in a prior enrichment processor. We would need to extend the connector with support for setting the attribute value from an OTTL expression.

**Review by axw — APPROVED** [review]

File: enrichments/trace/internal/elastic/span.go (hunk 1 above)
Comment (lahsivjar) [reply enrichments/trace/internal/elastic/span.go:186]: > we could use the count connector with an OTTL-based condition on parent_span_id being empty

Didn't get the count connector part since it doesn't support histograms, did you mean the span metrics connector? Also, in addition to checking for `parent_span_id` being empty, we would also need to check for remote span right?

Comment (axw) [reply enrichments/trace/internal/elastic/span.go:186]: I meant theoretically, if the count connector did support histograms, we _could_ use it to produce these metrics with configuration. Something like the following (conceptually):

```yaml
  count:
    spans:
      transaction:
        description: APM transaction aggregated metrics
        conditions:
        - attributes["transaction.name"] != ""
        attributes:
        - key: transaction.root
          ottl_expression: parent_span_id == ""
        - key: transaction.name
        - key: transaction.result
        - key: event.outcome
```

> Also, in addition to checking for parent_span_id being empty, we would also need to check for remote span right?

Yes you're right, for the condition we need to check if it's a transaction, and for the value of `transaction.root` we would check the value of `parent_span_id`.
